File: tests/lang-test/csharp/Models/Product.cs
// ripex-lang-test: C# Product — record, enum, interface, method.
namespace Ripex.Models
{
    public enum Category
    {
        Electronics,
        Clothing
    }

    public interface IPricable
    {
        double PriceOf();
    }

    public record Product(int Id, string Name, double Price, Category Cat) : IPricable
    {
        public double PriceOf() => Price;
    }
}

File: tests/lang-test/go/models/product.go
package models

// ripex-lang-test: Go Product model — interface, embedded struct.
type Category string

const (
	Electronics Category = "electronics"
	Clothing    Category = "clothing"
)

type Product struct {
	ID     int
	Name   string
	Price  float64
	Cat    Category
}

type Pricable interface {
	PriceOf() float64
}

func (p Product) PriceOf() float64 {
	return p.Price
}
